### PR TITLE
update to the latest dependencies

### PR DESCRIPTION
**Pull request by koushiro**
Signed-off-by: koushiro <koushiro.cqx@gmail.com>

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -10,12 +10,12 @@ repository = "https://github.com/tikv/yatp/"
...
-dashmap = "3.5.1"
-fail = "0.3"
+dashmap = "3.11.10"
+fail = "0.4"
 lazy_static = "1"
 num_cpus = "1.0"
-parking_lot_core = "0.7"
-prometheus = "0.8"
+parking_lot_core = "0.8"
+prometheus = { version = "0.10", default-features = false }
```
Comment (BusyJay) [inline Cargo.toml:18]: Can you also send the PR to TiKV? It seems to still use 0.8.

Comment (koushiro) [reply Cargo.toml:18]: Okay

Comment (koushiro) [reply Cargo.toml:18]: I sent a PR to TiKV several days ago, if there is no other problems, we can merge this PR.

**Review by BusyJay — APPROVED** [review]

**Comment by BusyJay** [comment]
@sticnarf PTAL

**Review by sticnarf — APPROVED** [review]
LGTM

**Pull request merged**